src/Parser/Shaman/Restoration/Modules/ShamanCore/HealingRainLocation.js

```diff
@@ -0,0 +1,119 @@
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import calculateEffectiveHealing from 'Parser/Core/calculateEffectiveHealing';
+
+const healingRainDiameter = 2200; // 10% margin of error
+
+/**
+ * Module to find the position of healing rain, and return how much extra healing spells modified by healing rain did.
+ * 
+ * The position is calculated by gathering all coordinates of people getting healed by rain, finding the extremes and treating it as an ellipse.
+ * Player position on heal is then compared against that.
+ */
+
+class HealingRainLocation extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  healingRainEvents = [];
+  newHealingRain = false;
+  lastHealingRainTick = 0;
+
+  on_initialized() {
+    const hasDeluge = this.combatants.selected.hasTalent(SPELLS.DELUGE_TALENT.id);
+    const hasRebalancers = this.combatants.selected.hasFeet(ITEMS.ELEMENTAL_REBALANCERS.id);
+    this.active = hasDeluge || hasRebalancers;
+  }
+
+  on_byPlayer_heal(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.HEALING_RAIN_HEAL.id && spellId !== SPELLS.RAINFALL.id) {
+      return;
+    }
+
+    // Pets with their insane movement speed (or blinking) messed the size up, so I decided to filter them out
+    const combatant = this.combatants.players[event.targetID];
+    if (!combatant) {
+      return;
+    }
+
+    if(this.newHealingRain) {
+      this.healingRainEvents.length = 0;
+      this.newHealingRain = false;
+    }
+
+    this.lastHealingRainTick = event.timestamp;
+    this.healingRainEvents.push(event);
+  }
+
+  on_byPlayer_begincast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.HEALING_RAIN_CAST.id || event.isCancelled) {
+      return;
+    }
+
+    // Setting a variable to reset the array on the next heal
+    // If we'd reset here it would sometimes clear the array before the healing is calculated
+    this.newHealingRain = true;
+  }
+
+  processLastRain(eventsDuringRain, healIncrease) {
```

Comment:
This method name is unclear and I think it does one too many things. How about splitting it in two; filterEventsInHealingRain which act like a filter for Array.prototype.filter and sumHealing which could perhaps be a reducer?